internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -73,24 +73,29 @@ spec:
             git_clone_and_checkout --repository $REPO --revision $REVISION
 
             for file in ${FILES}; do
+                # in case the file has comments, empty newlines and other special chages in the
```

Comment:
That's right. The thing is that the  yq's `|line`  function  removes all whitespaces until the first valid json, but keeps comments and `---`. After it, it preserves all of them. Notice that if you do `yq . <file>`  the output does not match this behavior as it removes only comments *after* the first valid json. At least this is the behavior in `v.4.34.1` which is the one in the release-utils image.

Reply:
Hmm, I'm pretty sure I added a comment here a few hours ago. Maybe I forgot to hit the Comment button? Weird.

> Notice that if you do yq . <file> the output does not match this behavior as it removes only comments after the first valid json.

I don't think it necessarily removes comments. See this snippet (it's the output of `yq . test.yaml` and the comment on the last line is preserved):

```yaml
---






resourceTemplates:
  - name: osd-fleet-manager-production
    parameters:
      CPU: [8, 10]
      MEMORY_LIMIT: 8Gi
    targets:
      - name: osd-fleet-manager-production-us-east-2
        # this is my comment
```

What I still don't understand is what this is for: `if(! match($0, "^#"))`, or I mean the whole `awk` command. I read it as a comment that starts at the beginning of the line. But it seems the `#` is actually a special character in the regex? Like `$` elsewhere? But I have a hard time finding it in the docs.

Reply:
Let-me clarify it in another way. 
Forget about the comments. The idea of this is: I need to know where the actual yaml block starts because `yq |line` will consider that spot as line 1. 
The "#" part is because we might have alphanumeric chars there, for instance`# blabla` which is not a valid start of yaml as  I noticed. Another case is that you might have `$schema` which is a valid word for start the yaml.

Once I know that is the start, I just get the position found -1 (which is the line above). So, from the start of the file, until there, all was ignored by `yq |line`.

Reply:
We discussed this later in a chat. It turns out that in some cases the behavior of `yq` can be inconsistent. If you add indented comments before the first key in the yaml, sometimes `yq` will count the line and sometimes not. But either way such comment is probably invalid yaml, because the indentation should be always aligned with some key/element.

Anyway, let's go with this for now and we can improve it later.